--- REQUIREMENTS.orig   2014-03-15 14:38:32.373531930 +0000
+++ REQUIREMENTS        2014-03-15 14:43:28.425826225 +0000
@@ -50,14 +50,18 @@
          http://libdbi.sourceforge.net/

 check_radius:
-       - Requires the radiusclient-ng library available from:
-         http://developer.berlios.de/projects/radiusclient-ng/
+        - Requires the freeradius-client library available from:
+         http://freeradius.org/freeradius-client/
+       - This plugin also works with the radiusclient-ng library available
+         from: http://developer.berlios.de/projects/radiusclient-ng/
+         Unfortunately this software package is unmaintained since many years.
        - This plugin also works with the original radiusclient library from
          ftp://ftp.cityline.net/pub/radiusclient/
                RPM (rpmfind): radiusclient 0.3.2, radiusclient-devel-0.3.2
          Unless you're using a distro-maintained version of this library you
-         probably want to use radiusclient-ng. The original radiusclient library is
-         unmaintained and has many known issues, particularly with 64bit systems.
+         probably want to use freeradius-client. The original radiusclient
+         library is unmaintained and has many known issues, particularly with
+         64bit systems.

 check_snmp:
        - Requires the NET-SNMP package available from
--- configure.ac.orig   2014-03-15 14:38:32.423530629 +0000
+++ configure.ac        2014-03-15 14:50:18.315141985 +0000
@@ -289,8 +289,15 @@
          RADIUSLIBS="-lradiusclient-ng"
       AC_SUBST(RADIUSLIBS)
     else
-      AC_MSG_WARN([Skipping radius plugin])
-      AC_MSG_WARN([install radius libs to compile this plugin (see REQUIREMENTS).])
+       AC_CHECK_LIB(freeradius-client,rc_read_config)
+       if test "$ac_cv_lib_freeradius_client_rc_read_config" = "yes"; then
+               EXTRAS="$EXTRAS check_radius"
+               RADIUSLIBS="-lfreeradius-client"
+               AC_SUBST(RADIUSLIBS)
+       else
+               AC_MSG_WARN([Skipping radius plugin])
+               AC_MSG_WARN([install radius libs to compile this plugin (see REQUIREMENTS).])
+       fi
     fi
   fi
   LIBS="$_SAVEDLIBS"
--- plugins/check_radius.c.orig 2014-03-14 11:14:13.954196455 +0000
+++ plugins/check_radius.c 2014-03-14 11:47:22.481635737 +0000
@@ -39,6 +39,9 @@
 #ifdef HAVE_LIBRADIUSCLIENT_NG
 #include <radiusclient-ng.h>
 rc_handle *rch = NULL;
+#elif HAVE_LIBFREERADIUS_CLIENT
+#include <freeradius-client.h>
+rc_handle *rch = NULL;
 #else
 #include <radiusclient.h>
 #endif
@@ -48,29 +51,41 @@
 void print_usage (void);

 /* libradiusclient(-ng) wrapper functions */
-#ifdef HAVE_LIBRADIUSCLIENT_NG
-#define my_rc_conf_str(a) rc_conf_str(rch,a)
-#define my_rc_send_server(a,b) rc_send_server(rch,a,b)
-#define my_rc_buildreq(a,b,c,d,e,f) rc_buildreq(rch,a,b,c,d,e,f)
-#define my_rc_own_ipaddress() rc_own_ipaddress(rch)
-#define my_rc_avpair_add(a,b,c,d) rc_avpair_add(rch,a,b,c,-1,d)
-#define my_rc_read_dictionary(a) rc_read_dictionary(rch, a)
-#else
+#ifdef HAVE_LIBRADIUSCLIENT
 #define my_rc_conf_str(a) rc_conf_str(a)
 #define my_rc_send_server(a,b) rc_send_server(a, b)
 #define my_rc_buildreq(a,b,c,d,e,f) rc_buildreq(a,b,c,d,e,f)
 #define my_rc_own_ipaddress() rc_own_ipaddress()
 #define my_rc_avpair_add(a,b,c,d) rc_avpair_add(a, b, c, d)
 #define my_rc_read_dictionary(a) rc_read_dictionary(a)
+#else
+#define my_rc_conf_str(a) rc_conf_str(rch,a)
+#define my_rc_send_server(a,b) rc_send_server(rch,a,b)
+#define my_rc_buildreq(a,b,c,d,e,f) rc_buildreq(rch,a,b,c,d,e,f)
+#define my_rc_own_ipaddress() rc_own_ipaddress(rch)
+#define my_rc_avpair_add(a,b,c,d) rc_avpair_add(rch,a,b,c,-1,d)
+#define my_rc_read_dictionary(a) rc_read_dictionary(rch, a)
 #endif

+
 /* REJECT_RC is only defined in some version of radiusclient. It has
  * been reported from radiusclient-ng 0.5.6 on FreeBSD 7.2-RELEASE */
 #ifndef REJECT_RC
 #define REJECT_RC BADRESP_RC
 #endif

-int my_rc_read_config(char *);
+int my_rc_read_config(char * a)
+ {
+
+#ifdef HAVE_LIBRADIUSCLIENT
+ return rc_read_config(a);
+#else
+ rch = rc_read_config(a);
+ return (rch == NULL) ? 1 : 0;
+#endif
+}
+
+

 char *server = NULL;
 char *username = NULL;
